test/controllers/user_role_controller_test.exs

```diff
@@ -60,66 +50,44 @@ defmodule CodeCorps.UserRoleControllerTest do
     test "creates and renders resource when data is valid", %{conn: conn} do
       user = insert(:user)
       role = insert(:role)
-
-      payload = build_payload |> put_relationships(user, role)
-      path = conn |> user_role_path(:create)
-      json = conn |> post(path, payload) |> json_response(201)
-
-      id = json["data"]["id"] |> String.to_integer
-      user_role = UserRole |> Repo.get(id)
-
-      assert json["data"]["id"] == "#{user_role.id}"
-      assert json["data"]["type"] == "user-role"
-      assert json["data"]["relationships"]["user"]["data"]["id"] == "#{user_role.user_id}"
-      assert json["data"]["relationships"]["role"]["data"]["id"] == "#{user_role.role_id}"
+      attrs = (%{user: user, role: role})
+      assert conn |> request_create(attrs) |> json_response(201)
     end
 
     @tag authenticated: :admin
     test "does not create resource and renders errors when data is invalid", %{conn: conn} do
-      payload = build_payload
-      path = conn |> user_role_path(:create)
-      json = conn |> post(path, payload) |> json_response(422)
-
-      assert json["errors"] != %{}
+      assert conn |> request_create(@invalid_attrs) |> json_response(422)
     end
 
     test "does not create resource and renders 401 when unauthenticated", %{conn: conn} do
-      path = conn |> user_role_path(:create)
-      assert conn |> post(path) |> json_response(401)
+      assert conn |> request_create(%{}) |> json_response(401)
     end
 
     @tag :authenticated
     test "does not create resource and renders 403 when not authorized", %{conn: conn} do
-      path = conn |> user_role_path(:create)
-      assert conn |> post(path, build_payload) |> json_response(403)
+      assert conn |> request_create(%{}) |>  json_response(403)
     end
   end
 
   describe "delete" do
     @tag authenticated: :admin
     test "deletes resource", %{conn: conn} do
       user_role = insert(:user_role)
-
-      path = conn |> user_role_path(:delete, user_role)
-      assert conn |> delete(path) |> response(204)
+      assert conn |> request_delete(user_role) |> response(204)
     end
```

Comment:
I think since request_delete doesn't have to have any params you can get rid of line 75 and the user_role params on 76, like you have for the other two delete tests